### PR TITLE
feat: [CDS-36613]: Add user group expression evaluation

### PR DESCRIPTION
**Pull request by Fernando-Dourado**
You can use the following comments to re-trigger PR Checks

- Compile: `trigger compile`
- runAeriformCheck: `trigger AeriformCheck`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/32644)

### CODE REVIEW

File: 400-rest/src/main/java/software/wings/sm/states/ApprovalState.java [hunk 1]
```diff
@@ -320,22 +327,49 @@ private void resolveUserGroupFromTemplate(ExecutionContext context, ExecutionCon
...
+    String renderedExpression = context.renderExpression(expression);
+
+    if (isEmpty(renderedExpression)) {
+      log.error("[EMPTY_EXPRESSION] Rendered expression is: [{}]. Original Expression: [{}], Context: [{}]",
+              renderedExpression, expression, context.asMap());
+      throw new InvalidRequestException("User group expression is invalid", USER);
+    }
+
+    List<String> userGroupNames =
+            Arrays.stream(renderedExpression.split(",")).map(String::trim).collect(Collectors.toList());
```
Comment (shivagowda) [inline 400-rest/src/main/java/software/wings/sm/states/ApprovalState.java:352]: so, splitting by "," always work? can you add an example of renderedExpression as comment above?

Comment (Fernando-Dourado) [reply 400-rest/src/main/java/software/wings/sm/states/ApprovalState.java:352]: The expression is like a variable ${myVariable} and it's content are set to "group1, group2". In the UI has an information pane mention to split values using comma.

**Comment by Fernando-Dourado** [comment]
trigger codebasehash

**Review by shivagowda — APPROVED** [review]
> Reviewed 2 of 4 files at r2, 2 of 2 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32644)* status: :shipit: complete! ✓

___
*[`400-rest/src/main/java/software/wings/sm/states/ApprovalState.java` line 352 at r1](https://reviewable.io/reviews/harness/harness-core/32644#-N0lvuZCXlDCkKunKNL6-r1-352:-N1LN49tBARLQARbwJLT:b-85i64v) ([raw file](https://github.com/harness/harness-core/blob/58fc88c3bb2745d1dccc70ecbba7706fbd40f296/400-rest/src/main/java/software/wings/sm/states/ApprovalState.java#L352)):*
<details><summary><i>Previously, Fernando-Dourado (Fernando Dourado) wrote…</i></summary><blockquote>

The expression is like a variable ${myVariable} and it's content are set to "group1, group2". In the UI has an information pane mention to split values using comma.
</blockquote></details>

LGTM

**Review by shivagowda — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32644)* status: :shipit: complete! ✓

**Review by shivagowda — APPROVED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32644#-:-N1LNbRcCFmbjjz94y09:bnfp4nl)* status: :shipit: complete! ✓

**Review by shivagowda — APPROVED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32644#-:-N1LNfd69xnexhg8gp8t:bnfp4nl)* status: :shipit: complete! ✓

**Review by shivagowda — APPROVED** [review]
> Reviewed 2 of 2 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32644)* status: :shipit: complete! ✓

**Review by abhinav-harness — APPROVED** [review]
> Reviewed 2 of 4 files at r2, 2 of 2 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32644)* status: :shipit: complete! ✓

**Pull request merged**